build/action.yaml

```diff
@@ -35,18 +40,49 @@ runs:
       uses: andstor/file-existence-action@v3
       with:
         files: yarn.lock
+    - name: Compute lockfile hash
+      if: inputs.s3-bucket-name != ''
+      id: lockfile_hash
+      run: echo "hash=${{ steps.check_yarn_lock.outputs.files_exists == 'true' && hashFiles('**/yarn.lock') || hashFiles('**/package-lock.json') }}" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Load cached node_modules if available
+      if: inputs.s3-bucket-name != ''
+      uses: everpcpc/actions-cache@v1
+      id: read_cache
+      with:
+        bucket: ${{ inputs.s3-bucket-name }}
+        use-fallback: false
+        path: node_modules
+        key: ${{ env.cache-name }}-${{ steps.lockfile_hash.outputs.hash }}
+        restore-keys: ${{ env.cache-name }}-
+      env:
+        AWS_REGION: ${{ inputs.s3-bucket-region }}
+        cache-name: ${{ github.event.repository.name }}-cache-node-modules
```

Comment:
cache entry is <repo-name>-cache-node-modules-<lockfile-hash>. Wasn't sure if we want to actually use the repo name as a directory, so <repo-name>/cache-node-modules-<lockfile-hash> for better organization in the s3 bucket

Reply:
Maybe `/cache-node-modules-` might be better for that?